enterprise/server/backends/userdb/userdb.go

```diff
@@ -698,6 +698,12 @@ func (d *UserDB) createUser(ctx context.Context, tx *db.DB, u *tables.User) erro
 		groupIDs = append(groupIDs, hydratedGroup.GroupID)
 	}
 
+	if u.UserID == "" {
+		return status.FailedPreconditionError("UserID is required")
+	}
+	if u.SubID == "" {
+		return status.FailedPreconditionError("SubID is required")
+	}
 	if u.Email == "" {
 		return status.FailedPreconditionErrorf("Auth token does not contain an email address")
 	}
```

Comment:
Could you move move these checks up before line 692 so they are the first thing that happens in this method?